doc/rtd/howto/launching.rst

```diff
@@ -0,0 +1,20 @@
+.. _launching:
```

Comment:
This is an index for different pages - each page describes how to launch cloud-init on a particular platform. Currently this only lists WSL, however we already have [a single page](https://docs.cloud-init.io/en/latest/howto/run_cloud_init_locally.html) which explains how to launch instances on various platforms: lxd, libvirt, multipass, and qemu. Given the overlap, I think that it would make sense to link from this page to each of those platforms - but we may want to split that page into individual ones to live next to wsl.rst.

@s-makin What do you think about that?

Reply:
I think that would make sense. Do you want me to do that as part of this PR or as a separate one?

Reply:
As part of this PR, if you don't mind.

Reply:
Done :)